modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaScreenContributorClassCheck.java

```diff
@@ -0,0 +1,159 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.source.formatter.parser.JavaClass;
+import com.liferay.source.formatter.parser.JavaClassParser;
+import com.liferay.source.formatter.parser.JavaMethod;
+import com.liferay.source.formatter.parser.JavaTerm;
+import com.liferay.source.formatter.parser.JavaVariable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Tamyris Bernardo
+ */
+public class UpgradeJavaScreenContributorClassCheck extends BaseUpgradeCheck {
+
+	@Override
+	protected String format(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		JavaClass javaClass = JavaClassParser.parseJavaClass(fileName, content);
+
+		String oldJavaClassContent = javaClass.getContent();
+
+		String newJavaClassContent = oldJavaClassContent;
+
+		Matcher matcher = _screenContributorPattern.matcher(
+			oldJavaClassContent);
+
+		if (matcher.find()) {
+			String javaTermContent = null;
+
+			List<String> oldVariables = new ArrayList<>();
+
+			List<String> oldMethods = new ArrayList<>();
+
+			for (JavaTerm childJavaTerm : javaClass.getChildJavaTerms()) {
+				if (childJavaTerm.isJavaVariable()) {
+					JavaVariable javaVariable = (JavaVariable)childJavaTerm;
+
+					javaTermContent = javaVariable.getContent();
+
+					oldVariables.add(javaTermContent);
+				}
+				else if (childJavaTerm.isJavaMethod()) {
+					JavaMethod javaMethod = (JavaMethod)childJavaTerm;
+
+					javaTermContent = javaMethod.getContent();
+
+					oldMethods.add(javaTermContent);
+				}
+			}
+
+			newJavaClassContent = _formatContent(
+				oldVariables, oldJavaClassContent, oldMethods, matcher);
+		}
+
+		return StringUtil.replace(
+			content, oldJavaClassContent, newJavaClassContent);
```

Comment:
@nicolas-sss, if there is no pattern matching then this replacement in unnecessary, can you please do it only if it's necessary?

Thanks